dashboard/app/models/user.rb

```diff
@@ -1981,6 +1981,13 @@ def undestroy
     restore(recursive: true, recovery_window: 5.minutes)
   end
 
+  def depended_upon_for_login?
+    return false unless teacher?
+    students.any? do |student|
+      student.can_create_personal_login? && student.sections_as_student.size == 1
```

Comment:
A student can be in multiple sections and still depend on a teacher for login, when all of the student's sections belong to the same teacher.  Our UI doesn't make this obvious, but it's possible by entering one's own section code in the "Move Students" dialog.

![screenshot from 2018-07-12 17-26-10](https://user-images.githubusercontent.com/1615761/42666300-1e63e25a-85f9-11e8-9da3-59aa15abd347.png)

![screenshot from 2018-07-12 17-28-05](https://user-images.githubusercontent.com/1615761/42666302-21983674-85f9-11e8-8426-5d53bc881929.png)

Reply:
good catch! i'll update the logic and add a unit test for this case